plugin/fulltext/mysql-test/ngram/ngram_token_size_basic.test

```diff
@@ -0,0 +1,50 @@
+
+#
+# show the global and session values;
+#
+call mtr.add_suppression("option 'ngram-token-size': signed value 0");
+call mtr.add_suppression("option 'ngram-token-size': signed value -1");
+call mtr.add_suppression("option 'ngram-token-size': signed value 11");
+
+select @@global.ngram_token_size;
+--error ER_INCORRECT_GLOBAL_LOCAL_VAR
+select @@session.ngram_token_size;
+show global variables like 'ngram_token_size';
+show session variables like 'ngram_token_size';
+--disable_warnings
+select * from information_schema.global_variables where variable_name='ngram_token_size';
+select * from information_schema.session_variables where variable_name='ngram_token_size';
+--enable_warnings
+
+# show it's read-only
+--error ER_INCORRECT_GLOBAL_LOCAL_VAR
+set global ngram_token_size = 3;
+
+# Valid values are from 1 to 10
+# ngram_token_size = 1 if we try to set ngram_token_size <= 0
+--let $restart_parameters= --ngram_token_size=0
+--source include/restart_mysqld.inc
+--disable_warnings
+select * from information_schema.global_variables where variable_name='ngram_token_size';
+--enable_warnings
+--let $restart_parameters= --ngram_token_size=-1
+--source include/restart_mysqld.inc
+--disable_warnings
+select * from information_schema.global_variables where variable_name='ngram_token_size';
+--enable_warnings
+# ngram_token_size = 10 if we try to set ngram_token_size >= 10
+--let $restart_parameters= --ngram_token_size=11
+--source include/restart_mysqld.inc
+--disable_warnings
+select * from information_schema.global_variables where variable_name='ngram_token_size';
+--enable_warnings
+# Boundry value 10
+--let $restart_parameters= --ngram_token_size=10
+--source include/restart_mysqld.inc
```

Comment:
I do not think that ‘boundry’ is a valid English word.

Where is the functional test? What if the index is created with one value of the token size, and we later change the size and restart the server? In which way will the index be corrupted? What kind of results would we see? Any crashes?

Shouldn’t the token size be persisted at the time of `CREATE FULLTEXT INDEX`? That is, a global value would only be the default setting for new indexes? Even better would be to make the token size part of the DDL syntax (`HA_IOPTION` would seem to be a good fit). I think that controlling the behaviour of DDL by global parameters (such as `innodb_file_per_table`) is a bad idea.

Reply:
> Where is the functional test? What if the index is created with one value of the token size, and we later change the size and restart the server?

Added a test where index is generated with 3-grams, but queried with 2-grams (in ngram_token_size.test).

> What kind of results would we see? Any crashes?

Results of the queries will be empty as if affected indexes were never populated. Will fix back again one token size is switched back to the value at which indexes were created.